### ISSUE TITLE
`evdns_base_resolve_ipv4` and `evdns_base_resolve_ipv6` do not take `/etc/hosts` into account

### ISSUE THREAD

**Issue by cvengler** [libevent/libevent#1765]
The `evdns_base_resolve_ipv4` and `evdns_base_resolve_ipv6` do not take the `/etc/hosts` file into account even if the respective `evdns_base` has been configured using `DNS_OPTION_HOSTSFILE`.

I believe this behavior should be either documented or preferably fixed, as resolving implies to me taking `/etc/hosts` into account. Preferably we could also add a flag to it, so `evdns_getaddrinfo` performs no double look-ups.

**Comment by SunnyPaprika** [comment]
Libevent uses /etc/hosts. You can step through `evdns_base_load_hosts_impl` and `evdns_base_parse_hosts_line` with a debugger and see if it behaves abnormally or exits to soon. The formatting of your host file could trigger something. https://github.com/libevent/libevent/blob/9a6a799e7002aae09d5e7cb244bdb913eed75a04/evdns.c#L5167

**Comment by azat** [comment]
Agree. @cvengler please provide a reproducer if it does not work for you.
